Library/Homebrew/utils/tty.rb

```diff
@@ -0,0 +1,51 @@
+module Tty
+  module_function
+
+  def strip_ansi(string)
+    string.gsub(/\033\[\d+(;\d+)*m/, "")
+  end
+
+  def width
+    `/usr/bin/tput cols`.strip.to_i
+  end
+
+  def truncate(string)
+    (w = width).zero? ? string.to_s : string.to_s[0, w - 4]
+  end
+
+  COLOR_CODES = {
+                  red:     31,
+                  green:   32,
+                  yellow:  33,
+                  blue:    34,
+                  magenta: 35,
+                  cyan:    36,
+                  default: 39,
+                }
+
+  STYLE_CODES = {
+                  reset:          0,
+                  bold:           1,
+                  italic:         3,
+                  underline:      4,
+                  strikethrough:  9,
+                  no_underline:  24,
+                }
+
+  CODES = COLOR_CODES.merge(STYLE_CODES)
+
+  CODES.each do |name, code|
+    define_singleton_method(name) do
+      @sequence = [*@sequence, code]
```

Comment:
I still think it would be good to split this into separate intent-signaling methods. I have my head around the `@sequence` stuff now but it's still a lot more complex to understand from the previous code. My preference is self-documenting code by splitting things into multiple methods that explain clearly what they do through their names. If you'd rather not do that: please add a lot more comments explaining what this is doing, how and why. Thanks!